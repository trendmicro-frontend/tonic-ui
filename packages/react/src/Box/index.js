import styled from '@emotion/styled';
import isPropValid from '@emotion/is-prop-valid';
import { ensureArray } from 'ensure-type';
import { cx, system } from '../shared/styled-system';

const shouldForwardProp = (() => {
  const stylePropMap = ensureArray(system.propNames)
    .reduce((acc, val) => {
      acc[val] = true;
      return acc;
    }, {});
  const omittedStylePropMap = {
    ...stylePropMap,

    // The `as` prop is supported by Emotion
    'as': true,
  };

  return prop => isPropValid(prop) && !omittedStylePropMap[prop];
})();

const createPseudoClassTransformFunction = name => prop => {
  const result = {};
  if (Array.isArray(prop)) {
    result[`${name}(${prop[0]})`] = prop[1];
  } else if (typeof prop === 'object') {
    for (const key in prop) {
      if (Object.prototype.hasOwnProperty.call(prop, key)) {
        result[`${name}(${key})`] = prop[key];
      }
    }
  }
  return result;
};

/**
 * Pseudo-classes
 * https://developer.mozilla.org/en-US/docs/Web/CSS/
 */

const active = [
  '&:active',
  '&[data-active=true]'
].join(',');
const checked = [
  '&[aria-checked=true]',
  '&:checked',
].join(',');
const selected = [
  '&[aria-selected=true]',
  '&[data-selected=true]',
  '&:active[data-selected=true]',
  '&:hover[data-selected=true]',
].join(',');
const disabled = [
  '&[aria-disabled=true]',
  '&:disabled',
  '&:disabled:hover',
  '&:disabled:focus',
  '&:hover[aria-disabled=true]',
  '&:focus[aria-disabled=true]',
].join(',');
const empty = '&:empty';
const enabled = [
  '&:enabled',
  '&:enabled:focus',
  '&:enabled:hover',
].join(',');
const firstChild = '&:first-child';
const firstOfType = '&:first-of-type';
const fullscreen = '&:fullscreen';
const focus = '&:focus';
const focusActive = '&:focus:active';
const focusHover = '&:focus:hover';
const focusWithin = '&:focus-within';
const focusSelected = ['&[aria-selected=true]:focus', '&[data-selected=true]:focus'].join(',');
const hover = '&:hover';
const indeterminate = '&:indeterminate';
const valid = [
  '&[aria-invalid=false]',
  '&:valid',
].join(',');
const invalid = [
  '&[aria-invalid=true]',
  '&:invalid',
].join(',');
const lastChild = '&:last-child';
const lastOfType = '&:last-of-type';
const nthOfTypeFn = createPseudoClassTransformFunction('&:nth-of-type');
const readOnly = [
  '&[aria-readonly=true]',
  '&:read-only',
].join(',');
const visited = '&:visited';

/**
 * Pseudo-elements
 * https://developer.mozilla.org/en-US/docs/Web/CSS/
 */

const after = '&::after';
const backdrop = '&::backdrop';
const before = '&::before';
const cue = '&::cue';
const firstLetter = '&::first-letter';
const firstLine = '&::first-line';
const placeholder = '&::placeholder';
const selection = '&::selection';

const _Box = styled('div', {
  shouldForwardProp,
})(system);

const Box = styled(_Box)(
  ({
    // pseudo-classes
    _active,
    _checked,
    _selected,
    _disabled,
    _empty,
    _enabled,
    _firstChild,
    _firstOfType,
    _fullscreen,
    _focus,
    _focusActive,
    _focusHover,
    _focusWithin,
    _focusSelected,
    _hover,
    _indeterminate,
    _valid,
    _invalid,
    _lastChild,
    _lastOfType,
    _nthOfType,
    _readOnly,
    _visited,

    // pseudo-elements
    __after,
    __backdrop,
    __before,
    __cue,
    __firstLetter,
    __firstLine,
    __placeholder,
    __selection,
  }) => {
    let rest = null;

    if (_nthOfType) {
      rest = { ...rest, ...nthOfTypeFn(_nthOfType) };
    }

    return cx({
      /**
       * Pseudo-classes must be declared in a specific order, as shown below:
       *
       * ```
       * :link
       * :visited
       * :hover
       * :active
       * ```
       *
       * Each pseudo-class corresponds to an event which can only happen later in the timeline than the one before.
       *
       * That is to say:
       *
       * 1. A link is unvisited before it is visited.
       * 2. A link is visited before it is hovered over.
       * 3. A link is hovered over before it is in active use.
       */
      [focus]: _focus,
      [visited]: _visited,
      [hover]: _hover,
      [focusHover]: _focusHover,
      [active]: _active,
      [focusActive]: _focusActive,
      [focusSelected]: _focusSelected,
      [focusWithin]: _focusWithin,
      [checked]: _checked,
      [selected]: _selected,
      [disabled]: _disabled,
      [empty]: _empty,
      [enabled]: _enabled,
      [firstChild]: _firstChild,
      [firstOfType]: _firstOfType,
      [fullscreen]: _fullscreen,
      [indeterminate]: _indeterminate,
      [valid]: _valid,
      [invalid]: _invalid,
      [lastChild]: _lastChild,
      [lastOfType]: _lastOfType,
      [readOnly]: _readOnly,

      // pseudo-elements
      [after]: __after,
      [backdrop]: __backdrop,
      [before]: __before,
      [cue]: __cue,
      [firstLetter]: __firstLetter,
      [firstLine]: __firstLine,
      [placeholder]: __placeholder,
      [selection]: __selection,

      ...rest
    });
  },
);

Box.displayName = 'Box';

export default Box;
