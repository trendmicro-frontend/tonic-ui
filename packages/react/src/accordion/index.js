import Accordion from './Accordion';
import AccordionBody from './AccordionBody';
import AccordionCollapse from './AccordionCollapse';
import AccordionHeader from './AccordionHeader';
import AccordionItem from './AccordionItem';
import AccordionToggle from './AccordionToggle';
import AccordionToggleIcon from './AccordionToggleIcon';
import useAccordion from './useAccordion';
import useAccordionItem from './useAccordionItem';

Accordion.Body = AccordionBody;
Accordion.Collapse = AccordionCollapse;
Accordion.Header = AccordionHeader;
Accordion.Item = AccordionItem;
Accordion.Toggle = AccordionToggle;
Accordion.Toggle.Icon = AccordionToggleIcon;

export {
  Accordion,
  AccordionBody,
  AccordionCollapse,
  AccordionHeader,
  AccordionItem,
  AccordionToggle,
  AccordionToggleIcon,
  useAccordion,
  useAccordionItem,
};
