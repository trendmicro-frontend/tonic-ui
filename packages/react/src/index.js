// components
export * from './accordion';
export * from './alert';
export * from './badge';
export * from './box';
export * from './button';
export * from './checkbox';
export * from './color-mode';
export * from './color-style';
export * from './css-baseline';
export * from './divider';
export * from './drawer';
export * from './flat-button';
export * from './flex';
export * from './grid';
export * from './icon';
export * from './image';
export * from './input';
export * from './link';
export * from './menu';
export * from './modal';
export * from './pagination';
export * from './popover';
export * from './popper';
export * from './portal';
export * from './presence';
export * from './radio';
export * from './search-input';
export * from './scrollbar';
export * from './select';
export * from './skeleton';
export * from './space';
export * from './spinner';
export * from './stack';
export * from './switch';
export * from './table';
export * from './tabs';
export * from './tag';
export * from './text';
export * from './textarea';
export * from './theme';
export * from './toast';
export * from './tooltip';
export * from './transitions';
export * from './visually-hidden';

// hooks
export useDisclosure from './deprecated/useDisclosure'; // deprecated

// settings
export colorStyle from './shared/color-style';
export theme from './shared/theme';
