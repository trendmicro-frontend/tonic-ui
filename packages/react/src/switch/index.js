import Switch from './Switch';
import ToggleSwitch from './ToggleSwitch';

export {
  Switch,
  ToggleSwitch, // deprecated
};
