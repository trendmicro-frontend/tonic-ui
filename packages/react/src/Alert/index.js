import React, { forwardRef } from 'react';
import Box from '../Box';
import ButtonBase from '../ButtonBase';
import Flex from '../Flex';
import Icon from '../Icon';
import Space from '../Space';
import {
  useAlertStyle,
  useAlertIconStyle,
  useAlertMessageStyle,
  useAlertCloseButtonStyle,
} from './styles';

const defaultVariant = 'solid';
const defaultSeverity = 'success';

const getIconBySeverity = (severity) => {
  const iconName = {
    success: 'success',
    info: 'info',
    warning: 'warning-triangle',
    error: 'error',
  }[severity];

  if (!iconName) {
    return null;
  }

  return (
    <Icon icon={`${iconName}`} />
  );
};

const AlertIcon = (props) => (
  <Flex {...props} />
);

const AlertMessage = (props) => (
  <Box {...props} />
);

const AlertCloseButton = (props) => (
  <ButtonBase {...props} />
);

const Alert = forwardRef((
  {
    isClosable: _isClosable = false,
    isCloseButtonVisible: LEGACY_isCloseButtonVisible = false, // eslint-disable-line camelcase
    onClose,
    variant = defaultVariant,
    severity = defaultSeverity,
    icon,
    children,
    ...rest
  },
  ref,
) => {
  const isClosable = _isClosable || LEGACY_isCloseButtonVisible; // eslint-disable-line camelcase
  const styleProps = useAlertStyle({ variant, severity });
  const iconStyleProps = useAlertIconStyle({ variant, severity });
  const messageStyleProps = useAlertMessageStyle();
  const closeButtonStyleProps = useAlertCloseButtonStyle({ variant });

  if (typeof icon === 'string') {
    icon = (<Icon icon={icon} />);
  }
  if (typeof icon === 'undefined') {
    icon = getIconBySeverity(severity);
  }

  return (
    <Flex
      ref={ref}
      align="flex-start"
      justify="space-between"
      {...styleProps}
      {...rest}
    >
      {!!icon && (
        <>
          <AlertIcon {...iconStyleProps}>
            {icon}
          </AlertIcon>
          <Space minWidth="2x" />
        </>
      )}
      <AlertMessage {...messageStyleProps}>
        {children}
      </AlertMessage>
      {!!isClosable && (
        <>
          <Space minWidth="4x" />
          <AlertCloseButton {...closeButtonStyleProps} onClick={onClose}>
            <Icon icon="close-s" />
          </AlertCloseButton>
        </>
      )}
    </Flex>
  );
});

Alert.displayName = 'Alert';

export default Alert;
