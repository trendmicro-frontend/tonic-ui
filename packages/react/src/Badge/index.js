import React, { forwardRef } from 'react';
import useBadgeStyle from './styles';
import Box from '../Box';

const Badge = forwardRef(
  (
    {
      variant = 'badge',
      variantColor = 'red',
      badgeContent,
      isHidden = false,
      children,
      offset,
      dotSize = 8,
      borderColor,
      borderWidth = 1,
      ...restProps
    },
    ref,
  ) => {
    const badgeBorderColor = borderColor ?? variantColor;
    const badgeStyleProps = useBadgeStyle({
      color: variantColor,
      hasChildren: !!children,
      showAsDot: variant === 'dot',
      offset,
      dotSize,
      borderColor: badgeBorderColor,
      borderWidth
    });

    return (
      <Box
        as="span"
        position="relative"
        display="inline-block"
      >
        {children}
        {!isHidden && (
          <Box
            ref={ref}
            as="span"
            fontWeight="normal"
            borderRadius="sm"
            whiteSpace="nowrap"
            {...badgeStyleProps}
            {...restProps}
          >
            {badgeContent}
          </Box>
        )}
      </Box>
    );
  }
);

Badge.displayName = 'Badge';

export default Badge;
