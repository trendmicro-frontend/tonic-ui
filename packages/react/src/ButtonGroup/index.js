import memoize from 'micro-memoize';
import React, { forwardRef } from 'react';
import Box from '../Box';
import { ButtonGroupProvider } from './context';

const getMemoizedState = memoize(state => ({ ...state }));

const ButtonGroup = forwardRef((
  {
    children,
    size = 'md',
    variant = 'default',
    orientation = 'horizontal',
    ...rest
  },
  ref
) => {
  const buttonGroupState = getMemoizedState({ size, variant, orientation });
  const orientationProps = {
    vertical: {
      flexDirection: 'column',
    },
    horizontal: {
      flexDirection: 'row',
    },
  }[orientation];
  return (
    <ButtonGroupProvider value={buttonGroupState}>
      <Box
        ref={ref}
        display="inline-flex"
        {...orientationProps}
        {...rest}
      >
        {children}
      </Box>
    </ButtonGroupProvider>
  );
});

ButtonGroup.displayName = 'ButtonGroup';

export default ButtonGroup;
