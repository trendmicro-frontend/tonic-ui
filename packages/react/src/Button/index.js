import React, { forwardRef } from 'react';
import ButtonBase from '../ButtonBase';
import { useButtonGroup } from '../ButtonGroup/context';
import { getButtonGroupCSS, useButtonStyle } from './styles';

const defaultSize = 'md';
const defaultVariant = 'default';

const Button = forwardRef(
  (
    {
      as: Comp = 'button',
      type = 'button',
      borderRadius = 'sm',
      size,
      variant,
      children,
      css,
      ...rest
    },
    ref,
  ) => {
    const buttonGroupContext = useButtonGroup();
    let isInGroup = false;
    let useVertical = false;
    if (buttonGroupContext) {
      isInGroup = true;
      const {
        size: buttonGroupSize,
        variant: buttonGroupVariant,
        orientation,
      } = { ...buttonGroupContext };
      useVertical = (orientation === 'vertical');
      // - Use the inherited value from the button group
      // - Fallback to the default value if the value is null or undefined
      size = (buttonGroupSize ?? size) ?? defaultSize;
      variant = (buttonGroupVariant ?? variant) ?? defaultVariant;
    } else {
      // Use the default value if the value is null or undefined
      size = size ?? defaultSize;
      variant = variant ?? defaultVariant;
    }
    const buttonStyleProps = useButtonStyle({
      size,
      variant,
      borderRadius,
    });
    css = [
      isInGroup && getButtonGroupCSS({ useVertical }),
      { ...css }
    ];

    return (
      <ButtonBase
        ref={ref}
        as={Comp}
        type={type}
        borderRadius={borderRadius}
        css={css}
        {...buttonStyleProps}
        {...rest}
      >
        { children }
      </ButtonBase>
    );
  },
);

Button.displayName = 'Button';

export default Button;
