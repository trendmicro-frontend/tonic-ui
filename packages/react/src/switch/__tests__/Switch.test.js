import { testA11y } from '@tonic-ui/react/test-utils/accessibility';
import { render } from '@tonic-ui/react/test-utils/render';
import { Switch } from '@tonic-ui/react/src';
import React, { useEffect, useRef } from 'react';

describe('Switch', () => {
  it('should render correctly', async () => {
    const renderOptions = {
      useCSSVariables: true,
    };
    const { container } = render((
      <>
        {/* Sizes */}
        <Switch size="sm" name="sizes">Label</Switch>
        <Switch size="md" name="sizes">Label</Switch>
        <Switch size="lg" name="sizes">Label</Switch>

        {/* States */}
        <Switch>Label</Switch>
        <Switch defaultChecked>Label</Switch>
        <Switch disabled>Label</Switch>
        <Switch disabled defaultChecked>Label</Switch>
      </>
    ), renderOptions);

    expect(container).toMatchSnapshot();

    await testA11y(container);
  });

  it('should render correctly with `inputRef` and `ref` props', () => {
    const TestComponent = () => {
      const ref = useRef();
      const inputRef = useRef();

      useEffect(() => {
        expect(inputRef.current.checked).toBe(true);

        // Note: The `ref` prop will be moved to the outermost element in the next major version
        expect(ref.current.checked).toBe(true);
      }, []);

      return (
        <Switch
          defaultChecked
          inputRef={inputRef}
          ref={ref}
        />
      );
    };

    render(
      <TestComponent />
    );
  });
});
