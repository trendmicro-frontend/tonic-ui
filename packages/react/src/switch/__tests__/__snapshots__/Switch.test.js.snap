// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should render correctly 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: top;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-2 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  border: 0;
  overflow: hidden;
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  white-space: nowrap;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 120ms;
  transition: all 120ms;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 38px;
  height: 22px;
}

input[type=checkbox]:focus+.emotion-4 [data-switch] [data-switch-track-halo] {
  fill: var(--tonic-colors-blue-40);
}

input[type=checkbox]:focus+.emotion-4 [data-switch] [data-switch-track-border] {
  fill: white;
}

input[type=checkbox]:hover:not(:disabled):not(:checked):not(:focus)+.emotion-4 [data-switch] [data-switch-track] {
  fill: var(--tonic-colors-gray-50);
}

input[type=checkbox]:disabled+.emotion-4 {
  opacity: 0.28;
}

input[type=checkbox]:checked:hover:not(:disabled):not(:focus)+.emotion-4 [data-switch] [data-switch-track] {
  fill: var(--tonic-colors-blue-30);
}

.emotion-4>* {
  opacity: 1;
}

input[type=checkbox]:checked+.emotion-4 [data-switch] [data-switch-track] {
  fill: var(--tonic-colors-blue-40);
}

input[type=checkbox]:checked+.emotion-4 [data-switch] [data-switch-thumb] {
  -webkit-transform: translateX(16px);
  -moz-transform: translateX(16px);
  -ms-transform: translateX(16px);
  transform: translateX(16px);
}

input[type=checkbox]:checked+.emotion-4>* {
  opacity: 1;
}

input[type=checkbox][data-indeterminate]+.emotion-4>* {
  opacity: 1;
}

.emotion-6 {
  width: 100%;
  height: 100%;
}

.emotion-8 {
  width: 38px;
  height: 22px;
  fill: none;
}

.emotion-10 {
  width: 34px;
  height: 18px;
  fill: none;
}

.emotion-12 {
  width: 32px;
  height: 16px;
  fill: var(--tonic-colors-gray-30);
  pointer-events: all;
}

.emotion-14 {
  fill: var(--tonic-colors-white-emphasis);
  -webkit-transform: translateX(0);
  -moz-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  transform-box: fill-box;
  -webkit-transition: -webkit-transform 250ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: transform 250ms cubic-bezier(0.4, 0, 0.2, 1);
}

.emotion-16 {
  margin-left: var(--tonic-space-2x);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  opacity: 1;
}

.emotion-22 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 120ms;
  transition: all 120ms;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 54px;
  height: 30px;
}

input[type=checkbox]:focus+.emotion-22 [data-switch] [data-switch-track-halo] {
  fill: var(--tonic-colors-blue-40);
}

input[type=checkbox]:focus+.emotion-22 [data-switch] [data-switch-track-border] {
  fill: white;
}

input[type=checkbox]:hover:not(:disabled):not(:checked):not(:focus)+.emotion-22 [data-switch] [data-switch-track] {
  fill: var(--tonic-colors-gray-50);
}

input[type=checkbox]:disabled+.emotion-22 {
  opacity: 0.28;
}

input[type=checkbox]:checked:hover:not(:disabled):not(:focus)+.emotion-22 [data-switch] [data-switch-track] {
  fill: var(--tonic-colors-blue-30);
}

.emotion-22>* {
  opacity: 1;
}

input[type=checkbox]:checked+.emotion-22 [data-switch] [data-switch-track] {
  fill: var(--tonic-colors-blue-40);
}

input[type=checkbox]:checked+.emotion-22 [data-switch] [data-switch-thumb] {
  -webkit-transform: translateX(24px);
  -moz-transform: translateX(24px);
  -ms-transform: translateX(24px);
  transform: translateX(24px);
}

input[type=checkbox]:checked+.emotion-22>* {
  opacity: 1;
}

input[type=checkbox][data-indeterminate]+.emotion-22>* {
  opacity: 1;
}

.emotion-26 {
  width: 54px;
  height: 30px;
  fill: none;
}

.emotion-28 {
  width: 50px;
  height: 26px;
  fill: none;
}

.emotion-30 {
  width: 48px;
  height: 24px;
  fill: var(--tonic-colors-gray-30);
  pointer-events: all;
}

.emotion-40 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 120ms;
  transition: all 120ms;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 70px;
  height: 38px;
}

input[type=checkbox]:focus+.emotion-40 [data-switch] [data-switch-track-halo] {
  fill: var(--tonic-colors-blue-40);
}

input[type=checkbox]:focus+.emotion-40 [data-switch] [data-switch-track-border] {
  fill: white;
}

input[type=checkbox]:hover:not(:disabled):not(:checked):not(:focus)+.emotion-40 [data-switch] [data-switch-track] {
  fill: var(--tonic-colors-gray-50);
}

input[type=checkbox]:disabled+.emotion-40 {
  opacity: 0.28;
}

input[type=checkbox]:checked:hover:not(:disabled):not(:focus)+.emotion-40 [data-switch] [data-switch-track] {
  fill: var(--tonic-colors-blue-30);
}

.emotion-40>* {
  opacity: 1;
}

input[type=checkbox]:checked+.emotion-40 [data-switch] [data-switch-track] {
  fill: var(--tonic-colors-blue-40);
}

input[type=checkbox]:checked+.emotion-40 [data-switch] [data-switch-thumb] {
  -webkit-transform: translateX(32px);
  -moz-transform: translateX(32px);
  -ms-transform: translateX(32px);
  transform: translateX(32px);
}

input[type=checkbox]:checked+.emotion-40>* {
  opacity: 1;
}

input[type=checkbox][data-indeterminate]+.emotion-40>* {
  opacity: 1;
}

.emotion-44 {
  width: 70px;
  height: 38px;
  fill: none;
}

.emotion-46 {
  width: 66px;
  height: 34px;
  fill: none;
}

.emotion-48 {
  width: 64px;
  height: 32px;
  fill: var(--tonic-colors-gray-30);
  pointer-events: all;
}

.emotion-90 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: top;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
}

.emotion-106 {
  margin-left: var(--tonic-space-2x);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  opacity: 0.28;
}

<div>
  <label
    class="emotion-0 emotion-1"
  >
    <input
      class="emotion-2 emotion-1"
      name="sizes"
      role="switch"
      type="checkbox"
      value=""
    />
    <div
      aria-hidden="true"
      class="emotion-4 emotion-1"
    >
      <svg
        class="emotion-6 emotion-1"
        data-switch="true"
        viewBox="0 0 38 22"
      >
        <rect
          class="emotion-8 emotion-1"
          data-switch-track-halo="true"
          rx="11"
          stroke-width="0"
          x="0"
          y="0"
        />
        <rect
          class="emotion-10 emotion-1"
          data-switch-track-border="true"
          rx="9"
          stroke-width="0"
          x="2"
          y="2"
        />
        <rect
          class="emotion-12 emotion-1"
          data-switch-track="true"
          rx="8"
          x="3"
          y="3"
        />
        <circle
          class="emotion-14 emotion-1"
          cx="11"
          cy="11"
          data-switch-thumb="true"
          r="6"
        />
      </svg>
    </div>
    <div
      class="emotion-16 emotion-1"
    >
      Label
    </div>
  </label>
  <label
    class="emotion-0 emotion-1"
  >
    <input
      class="emotion-2 emotion-1"
      name="sizes"
      role="switch"
      type="checkbox"
      value=""
    />
    <div
      aria-hidden="true"
      class="emotion-22 emotion-1"
    >
      <svg
        class="emotion-6 emotion-1"
        data-switch="true"
        viewBox="0 0 54 30"
      >
        <rect
          class="emotion-26 emotion-1"
          data-switch-track-halo="true"
          rx="15"
          stroke-width="0"
          x="0"
          y="0"
        />
        <rect
          class="emotion-28 emotion-1"
          data-switch-track-border="true"
          rx="13"
          stroke-width="0"
          x="2"
          y="2"
        />
        <rect
          class="emotion-30 emotion-1"
          data-switch-track="true"
          rx="12"
          x="3"
          y="3"
        />
        <circle
          class="emotion-14 emotion-1"
          cx="15"
          cy="15"
          data-switch-thumb="true"
          r="9"
        />
      </svg>
    </div>
    <div
      class="emotion-16 emotion-1"
    >
      Label
    </div>
  </label>
  <label
    class="emotion-0 emotion-1"
  >
    <input
      class="emotion-2 emotion-1"
      name="sizes"
      role="switch"
      type="checkbox"
      value=""
    />
    <div
      aria-hidden="true"
      class="emotion-40 emotion-1"
    >
      <svg
        class="emotion-6 emotion-1"
        data-switch="true"
        viewBox="0 0 70 38"
      >
        <rect
          class="emotion-44 emotion-1"
          data-switch-track-halo="true"
          rx="19"
          stroke-width="0"
          x="0"
          y="0"
        />
        <rect
          class="emotion-46 emotion-1"
          data-switch-track-border="true"
          rx="17"
          stroke-width="0"
          x="2"
          y="2"
        />
        <rect
          class="emotion-48 emotion-1"
          data-switch-track="true"
          rx="16"
          x="3"
          y="3"
        />
        <circle
          class="emotion-14 emotion-1"
          cx="19"
          cy="19"
          data-switch-thumb="true"
          r="12"
        />
      </svg>
    </div>
    <div
      class="emotion-16 emotion-1"
    >
      Label
    </div>
  </label>
  <label
    class="emotion-0 emotion-1"
  >
    <input
      class="emotion-2 emotion-1"
      role="switch"
      type="checkbox"
      value=""
    />
    <div
      aria-hidden="true"
      class="emotion-22 emotion-1"
    >
      <svg
        class="emotion-6 emotion-1"
        data-switch="true"
        viewBox="0 0 54 30"
      >
        <rect
          class="emotion-26 emotion-1"
          data-switch-track-halo="true"
          rx="15"
          stroke-width="0"
          x="0"
          y="0"
        />
        <rect
          class="emotion-28 emotion-1"
          data-switch-track-border="true"
          rx="13"
          stroke-width="0"
          x="2"
          y="2"
        />
        <rect
          class="emotion-30 emotion-1"
          data-switch-track="true"
          rx="12"
          x="3"
          y="3"
        />
        <circle
          class="emotion-14 emotion-1"
          cx="15"
          cy="15"
          data-switch-thumb="true"
          r="9"
        />
      </svg>
    </div>
    <div
      class="emotion-16 emotion-1"
    >
      Label
    </div>
  </label>
  <label
    class="emotion-0 emotion-1"
  >
    <input
      checked=""
      class="emotion-2 emotion-1"
      role="switch"
      type="checkbox"
      value=""
    />
    <div
      aria-hidden="true"
      class="emotion-22 emotion-1"
    >
      <svg
        class="emotion-6 emotion-1"
        data-switch="true"
        viewBox="0 0 54 30"
      >
        <rect
          class="emotion-26 emotion-1"
          data-switch-track-halo="true"
          rx="15"
          stroke-width="0"
          x="0"
          y="0"
        />
        <rect
          class="emotion-28 emotion-1"
          data-switch-track-border="true"
          rx="13"
          stroke-width="0"
          x="2"
          y="2"
        />
        <rect
          class="emotion-30 emotion-1"
          data-switch-track="true"
          rx="12"
          x="3"
          y="3"
        />
        <circle
          class="emotion-14 emotion-1"
          cx="15"
          cy="15"
          data-switch-thumb="true"
          r="9"
        />
      </svg>
    </div>
    <div
      class="emotion-16 emotion-1"
    >
      Label
    </div>
  </label>
  <label
    class="emotion-90 emotion-1"
  >
    <input
      class="emotion-2 emotion-1"
      disabled=""
      role="switch"
      type="checkbox"
      value=""
    />
    <div
      aria-hidden="true"
      class="emotion-22 emotion-1"
    >
      <svg
        class="emotion-6 emotion-1"
        data-switch="true"
        viewBox="0 0 54 30"
      >
        <rect
          class="emotion-26 emotion-1"
          data-switch-track-halo="true"
          rx="15"
          stroke-width="0"
          x="0"
          y="0"
        />
        <rect
          class="emotion-28 emotion-1"
          data-switch-track-border="true"
          rx="13"
          stroke-width="0"
          x="2"
          y="2"
        />
        <rect
          class="emotion-30 emotion-1"
          data-switch-track="true"
          rx="12"
          x="3"
          y="3"
        />
        <circle
          class="emotion-14 emotion-1"
          cx="15"
          cy="15"
          data-switch-thumb="true"
          r="9"
        />
      </svg>
    </div>
    <div
      class="emotion-106 emotion-1"
    >
      Label
    </div>
  </label>
  <label
    class="emotion-90 emotion-1"
  >
    <input
      checked=""
      class="emotion-2 emotion-1"
      disabled=""
      role="switch"
      type="checkbox"
      value=""
    />
    <div
      aria-hidden="true"
      class="emotion-22 emotion-1"
    >
      <svg
        class="emotion-6 emotion-1"
        data-switch="true"
        viewBox="0 0 54 30"
      >
        <rect
          class="emotion-26 emotion-1"
          data-switch-track-halo="true"
          rx="15"
          stroke-width="0"
          x="0"
          y="0"
        />
        <rect
          class="emotion-28 emotion-1"
          data-switch-track-border="true"
          rx="13"
          stroke-width="0"
          x="2"
          y="2"
        />
        <rect
          class="emotion-30 emotion-1"
          data-switch-track="true"
          rx="12"
          x="3"
          y="3"
        />
        <circle
          class="emotion-14 emotion-1"
          cx="15"
          cy="15"
          data-switch-thumb="true"
          r="9"
        />
      </svg>
    </div>
    <div
      class="emotion-106 emotion-1"
    >
      Label
    </div>
  </label>
</div>
`;
